fifeforspark/base_modelers.py

```diff
@@ -0,0 +1,430 @@
+        data = data.drop('True mask')
+        return subset
+    return subset
+
+
+def compute_metrics_for_binary_outcomes(
+    actuals: pyspark.sql.DataFrame, predictions: pyspark.sql.DataFrame,
+    threshold_positive: Union[None, str, float] = 0.5, share_positive: Union[None, str, float] = None
+) -> OrderedDict:
+    actuals = actuals.select(actuals['actuals'].cast(DoubleType()))
+    num_true = actuals.agg({'actuals': 'sum'}).first()[0]
+    total = actuals.count()
+    metrics = OrderedDict()
+    if (num_true > 0) & (num_true < total):
+        predictions = predictions.withColumn(
+            "row_id", monotonically_increasing_id())
+        actuals = actuals.withColumn("row_id", monotonically_increasing_id())
+        preds_and_labs = predictions.join(actuals, predictions.row_id == actuals.row_id).select(
+            predictions.predictions, actuals.actuals)
+        scoresAndLabels = preds_and_labs.rdd.map(tuple)
+        evaluator = BinaryClassificationMetrics(scoresAndLabels)
+        metrics['AUROC'] = evaluator.areaUnderROC
+
+    else:
+        metrics["AUROC"] = np.nan
+
+    # Difficult to do a weighted avg in pyspark, leaving out for now
+    mean_predict = predictions.agg({'predictions': 'mean'}).first()[0]
+    mean_actual = actuals.agg({'actuals': 'mean'}).first()[0]
+    metrics["Predicted Share"] = mean_predict
+    metrics["Actual Share"] = mean_actual
+    # Checks if actuals doesn't have a single row, if so then it's not empty
+    if actuals.first() is not None:
+        if share_positive == 'predicted':
+            share_positive = predictions.agg(
+                {'predictions': 'mean'}).first()[0]
+        if share_positive is not None:
+            threshold_positive = predictions.approxQuantile(
+                'predictions', [1-share_positive], relativeError=.1)[0]
+        elif threshold_positive == 'predicted':
+            threshold_positive = mean_predict
+
+        preds_and_labs = preds_and_labs.withColumn('predictions', when(
+            preds_and_labs.predictions >= threshold_positive, 1).otherwise(0))
+        TP = preds_and_labs.select(((preds_and_labs.predictions == 1) & (
+            preds_and_labs.actuals == 1)).cast('int').alias('TP')).agg({'TP': 'sum'}).first()[0]
+        TN = preds_and_labs.select(((preds_and_labs.predictions == 0) & (
+            preds_and_labs.actuals == 0)).cast('int').alias('TN')).agg({'TN': 'sum'}).first()[0]
+        FP = preds_and_labs.select(((preds_and_labs.predictions == 1) & (
+            preds_and_labs.actuals == 0)).cast('int').alias('FP')).agg({'FP': 'sum'}).first()[0]
+        FN = preds_and_labs.select(((preds_and_labs.predictions == 0) & (
+            preds_and_labs.actuals == 1)).cast('int').alias('FN')).agg({'FN': 'sum'}).first()[0]
+
+        metrics["True Positives"] = TP
+        metrics["False Negatives"] = FN
+        metrics["False Positives"] = FP
+        metrics["True Negatives"] = TN
+
+    return metrics
```

Comment:
dictionary_outcomes = {'True Positives': [1,0], 'True Negatives': [0,0], 'False Positives': [0,1], 'False Negatives': [1,1]}
for key in dictionary_outcomes.keys():
  metrics[key] = preds_and_labs.select(((preds_and_labs.predictions == dictionary_outcomes[key][0]) & (
            preds_and_labs.actuals == dictionary_outcomes[key][1])).cast('int').alias('A')).agg({'A': 'sum'}).first()[0]

Something like this